From 4d687d8434fc4ed10182a26c1399dbf0fff6f8cc Mon Sep 17 00:00:00 2001
From: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
Date: Mon, 19 Apr 2021 18:47:50 +0200
Subject: [PATCH] ovs-vswitchd.service: create vm-sockets directory

Create a 'vm-sockets' directory to store the sockets used to communicate
between openvswitch and qemu.
This directory is owned by qemu to make sure it can create and write
into the sockets. The sgid sticky bit is enabled to ensure openvswitch
user keeps write access to the sockets.

Signed-off-by: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
---
 rhel/usr_lib_systemd_system_ovs-vswitchd.service.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
index e4ea2f784..58a5182a0 100644
--- a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
+++ b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in
@@ -21,6 +21,9 @@ LimitSTACK=2M
 ExecStartPre=-/bin/sh -c '/bin/chown :$${OVS_USER_ID##*:} /dev/hugepages'
 ExecStartPre=-/bin/chmod 0775 /dev/hugepages
 @end_dpdk@
+ExecStartPre=-/bin/mkdir /var/run/openvswitch/vm-sockets
+ExecStartPre=+/bin/chown -R qemu /var/run/openvswitch/vm-sockets
+ExecStartPre=+/bin/chmod g+ws /var/run/openvswitch/vm-sockets
 ExecStart=/usr/share/openvswitch/scripts/ovs-ctl \
           --no-ovsdb-server --no-monitor --system-id=random \
           ${OVS_USER_OPT} \
-- 
2.17.1

